clang/test/CodeGenSYCL/matrix.cpp

```diff
@@ -32,3 +33,9 @@ void f5(__spv::__spirv_JointMatrixINTEL<sycl::bfloat16, 10, 2, 0, 0> *matrix) {}
 
 // CHECK: @_Z2f6{{.*}}(%spirv.JointMatrixINTEL._i128_10_2_0_0
 void f6(__spv::__spirv_JointMatrixINTEL<_BitInt(128), 10, 2, 0, 0> *matrix) {}
+
+// CHECK: @_Z2f7{{.*}}(%spirv.JointMatrixINTEL._tf32_10_2_0_0
```

Comment:
The patch also adds a check for tf32, so it's addressed in the test.

Though, I have no idea how to write a test for the original issue.

Reply:
I assume the test on line 13 isn't it?

Reply:
I'm not sure if the solution here is to manually check types like this or figure out what is wrong with `print`

Reply:
Based on an offline discussion,  manual checks added instead of print is better, since it restricts the  base type of matrix to 'allowed' types.